beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -771,45 +763,48 @@ func (vs *Server) packAttestations(ctx context.Context, latestState state.Beacon
 		return nil, errors.Wrap(err, "could not filter attestations")
 	}
 
-	// If there is any room left in the block, consider unaggregated attestations as well.
-	numAtts := uint64(len(atts))
-	if numAtts < params.BeaconConfig().MaxAttestations {
-		uAtts, err := vs.AttPool.UnaggregatedAttestations()
-		if err != nil {
-			return nil, errors.Wrap(err, "could not get unaggregated attestations")
-		}
-		uAtts, err = vs.filterAttestationsForBlockInclusion(ctx, latestState, uAtts)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not filter attestations")
-		}
-		atts = append(atts, uAtts...)
+	uAtts, err := vs.AttPool.UnaggregatedAttestations()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get unaggregated attestations")
+	}
+	uAtts, err = vs.filterAttestationsForBlockInclusion(ctx, latestState, uAtts)
```

Comment:
Since we refactored `filterAttestationsForBlockInclusion`, let's make the method name more specific

```suggestion
	uAtts, err = vs.validateAndDeleteAttsInPool(ctx, latestState, uAtts)
```